types/pb/sf/ethereum/type/v2/type.go

```diff
@@ -201,8 +205,23 @@ func callAtIndex(idx uint32, calls []*Call) *Call {
 	return nil
 }
 
+type Variant int
+
+const (
+	VariantUnset Variant = iota
+	VariantGeth
+	VariantPolygon
+	VariantBNB
+)
+
 // NormalizeBlockInPlace
-func (block *Block) NormalizeInPlace() {
+func (block *Block) NormalizeInPlace(v Variant) {
+
+	switch v {
+	case VariantPolygon:
+		block.TransactionTraces = CombinePolygonSystemTransactions(block.TransactionTraces, block.Number, block.Hash)
```

Comment:
Let's wrap that in a if that checks if system transactions are present:
```suggestion
if hasPolygonSystemTransactions(block) {
		block.TransactionTraces = CombinePolygonSystemTransactions(block.TransactionTraces, block.Number, block.Hash)
		}
```